src/utils.h

```diff
@@ -0,0 +1,41 @@
+/****************************************************************************
+ * Copyright (c) 2018-2020 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef UTIL_H
+#define UTIL_H
+
+#include <iostream>
+
+#include <mpi.h>
+
+bool print_rank();
+
+template <typename t_last>
+void print_error( t_last &&last )
+{
+    if ( print_rank() )
+    {
+        std::cout << last << std::endl
+                  << "Aborting after error." << std::endl
+                  << std::endl;
+        MPI_Abort( MPI_COMM_WORLD, 1 );
```

Comment:
Is this a collective? I guess that is always one of the issues of throwing exceptions in MPI code, you can get a hang because the exception is not a collective and therefore you can't always find it. Is it possible that `print_error` will only get called by one rank or something?

Reply:
My only intention was to use one way to exit consistently. This will only ever be called by one rank, so maybe just using `throw` is fine

Reply:
In that case I would `throw` then - better to be consistent with C++ if possible for code not on the device. Easy to catch exceptions too in a debugger.

Reply:
Agreed

Reply:
Done